src/shogun/multiclass/tree/CARTree.cpp

```diff
@@ -491,10 +518,23 @@ std::shared_ptr<BinaryTreeMachineNode<CARTreeNodeData>> CARTree::CARTtrain(std::
 	right_child->data.transit_into_values=right_transit;
 	node->data.num_leaves=left_child->data.num_leaves+right_child->data.num_leaves;
 	node->data.weight_minus_branch=left_child->data.weight_minus_branch+right_child->data.weight_minus_branch;
-
+	node->data.impurity = node_impurity;
 	return node;
 }
-
+SGVector<float64_t> CARTree::get_feature_importance()
```

Comment:
mmm here this function *always* recomputes the feature importance as far as i understand. since `compute_feature_importance` is called here... could we call this on the end of the train? i mean the feature importance doesn't change after a train.

so basically what i would do is simply have a void function that is called after the train is done, it updates the internal `m_feature_importances`, and the CARTree::get_feature_importance simply returns that SGVector...

Reply:
and then i wouldn't add a watch_method but simply do an SG_ADD for the m_feature_importances (see other examples in the codebase)

Reply:
yes, the way of calling `compute_feature_importance` after the train is better, i will change it